### ISSUE TITLE
Painfully slow server performance on ASP.NET Core 1.0.1 in IIS

### ISSUE THREAD

**Issue by TheDenys** [aspnet/IISIntegration#286]
ASP.NET Core application targeting .NET 4.5.2 is performing very poorly on Windows Server 2012R2 running under IIS.

**Comment by rynowak** [comment]
Can you share more information about what you mean by *performing very poorly*? Specifically what kind of performance data are you looking at.

 I'm afraid we can't investigate this without an example of the app in question and a description of what the client is doing.

**Comment by TheDenys** [comment]
Every page takes ~30 seconds to load. I left the About page pretty much untouched from the template generated by Visual Studio, end even it loads about 30 seconds.
First page load takes even longer - ~45 seconds.

Here's web config:
`<?xml version="1.0" encoding="utf-8"?>
<configuration>
  
  <system.webServer>
    <handlers>
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="true" />
  </system.webServer>
</configuration>`

ConfigureServices:
`        public void ConfigureServices(IServiceCollection services)
        {
            services.AddSingleton<IEntitlementsProvider, EntitlementsProvider>();

            services
                .AddMvc(config =>
                {
                    config.RespectBrowserAcceptHeader = true;
                    config.InputFormatters.Add(new XmlSerializerInputFormatter());
                    config.OutputFormatters.Add(new XmlSerializerOutputFormatter());
                })
                .AddJsonOptions(jsonOptions =>
                {
                    jsonOptions.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
                });
        }
`

Project json:
`{
  "dependencies": {
    "Microsoft.AspNetCore.Diagnostics": "1.0.0",
    "Microsoft.AspNetCore.Mvc": "1.0.1",
    "Microsoft.AspNetCore.Mvc.Formatters.Json": "1.0.1",
    "Microsoft.AspNetCore.Mvc.Formatters.Xml": "1.0.1",
    "Microsoft.AspNetCore.Razor.Tools": {
        "version": "1.0.0-preview2-final",
        "type": "build"
    },
    "Microsoft.AspNetCore.Routing": "1.0.1",
    "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0",
    "Microsoft.AspNetCore.Server.Kestrel": "1.0.1",
    "Microsoft.AspNetCore.StaticFiles": "1.0.0",
    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0",
    "Microsoft.Extensions.Configuration.Json": "1.0.0",
    "Microsoft.Extensions.Logging": "1.0.0",
    "Microsoft.Extensions.Logging.Console": "1.0.0",
    "Microsoft.Extensions.Logging.Debug": "1.0.0",
    "Microsoft.Extensions.Options.ConfigurationExtensions": "1.0.0",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0"
  },

  "tools": {
    "BundlerMinifier.Core": "2.0.238",
    "Microsoft.AspNetCore.Razor.Tools": "1.0.0-preview2-final",
    "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.0.0-preview2-final"
  },

  "frameworks": {
    "net452": {}
  },

  "buildOptions": {
    "emitEntryPoint": true,
    "preserveCompilationContext": true
  },

  "publishOptions": {
    "include": [
      "wwwroot",
      "**/*.cshtml",
      "appsettings.json",
      "web.config"
    ]
  },

  "scripts": {
    //"prepublish": [ "bower install", "dotnet bundle" ],
    "postpublish": [ "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%" ]
  }
}
`

**Comment by TheDenys** [comment]
Hi Ryan,
The super slow page load happens only when i visit pages from the **same machine**.
I tried to access web pages and WebAPI endpoints from a remote machine and response time is just normal (under 1 second). I tried on 2 identical machines with Server 2012R2. I was not able to gain W2K8 machine for tests. On development machine with Win7 x64 the site performs with no issues. Looks like a specific environmental problem.
Thanks for fast replies. Feel free to close the ticket.
Cheers,
Denis.
